apps/dashboard/src/modules/user/components/UserSettingsComponent.vue

```diff
@@ -80,6 +87,58 @@ import { useConfirm } from "primevue/useconfirm";
 import { useToast } from "primevue/usetoast";
 import apiService from "@/services/ApiService";
 import { handleError } from "@/utils/errorUtils";
+import { useUserStore } from "@sudosos/sudosos-frontend-common";
+
+const status: Ref<string> = ref("");
+
+async function startScan() {
+  try {
+    if (!('NDEFReader' in window)) {
+      status.value = "Web NFC is not supported on this device.";
+      console.log('no');
+      toast.add({
+            severity: "error",
+            summary: "Unsupported",
+            detail: "Browser is not mobile and does not have Web NFC.",
+            life: 3000,
+          });
+    } else {
+      const ndef = new (window as any).NDEFReader();
+      await ndef.scan();
+      toast.add({
+        severity: "info",
+        summary: "Scan started",
+        detail: "NFC Scan started, please hold your NFC card to your phone!",
+        life: 3000,
+      });
```

Comment:
Prefer to put the `success` toast in a `.then` to make sure its only shown if it actually works [read more about promises here](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise)

Reply:
done